JoinColony/colonyCDapp
# fix: handle loading state for c2f components

## Description

![loading](https://github.com/user-attachments/assets/9073112a-2b56-4c56-a159-dcaf19282a4b)

## Testing

> [!IMPORTANT]
> - Open the `amplify/backend/function/bridgeXYZMutation/src/index.js` file and make sure you enter the right sandbox api key:
> ```js
> // Use the sandbox API key instead of 'xxx'
> let apiKey = 'xxx';
> ```
> - Create a `.env.local.secrets` file in the parent directory and populate it with:
> ```js
> BSCSCAN_API_KEY=
> ETHERSCAN_API_KEY=
> GOOGLE_TAG_MANAGER_ID=
> PINATA_API_KEY=
> PINATA_API_SECRET=
> COINGECKO_API_KEY=
> COINGECKO_API_URL=
> POSTHOG_KEY=
> POSTHOG_HOST=
> USDC_LOCAL_ADDRESS=
> ```
> - You will need to provide values for `POSTHOG_KEY` and `POSTHOG_HOST` so please ask a fellow dev about this
> - Run `npm run prepare` to inject these vars
> - Run `npm frontend`

### Testing steps
> [!NOTE]
> The kyc request is too fast when you have no data so you won't see the loader, so we're gonna have to complete a KYC flow

1. Visit the following URL, it should take you directly to the Crypto to Fiat tab: http://localhost:9091/account/crypto-to-fiat
2. Complete Step #1 Verification
3. Once you're approved, just reload the page
4. Check that you can see the loading states

## Diffs

**New stuff** ✨

* A new HoC component that renders a configurable Skeleton loader when you give it a truthy value for its `isLoading` prop

**Changes** 🛠️

* The page is now using the `loading` state from the `useCheckKycStatusMutation` GraphQL call to detect if the request is busy

Resolves #2747
